test/utils/SpokePoolUtils.ts

```diff
@@ -68,3 +103,669 @@ export function v3FillFromDeposit(deposit: V3DepositWithBlock, relayer: string):
+  await spokePool.connect(relayer).fillRelayWithUpdatedDeposit(
+    ...getV2FillRelayUpdatedFeeParams(
+      relayDataFromFill,
+      fillToBuildFrom.amount
+        .mul(toBNWei(1).sub(fillToBuildFrom.realizedLpFeePct.add(updatedRelayerFeePct)))
+        .mul(toBNWei(pctOfDepositToFill))
+        .div(toBNWei(1))
+        .div(toBNWei(1)),
+      updatedRelayerFeePct,
+      signature,
+      Number(relayDataFromFill.destinationChainId),
+      newRecipient ?? relayDataFromFill.recipient,
+      newMessage ?? relayDataFromFill.message
+    )
+  );
+  const [events, destinationChainId] = await Promise.all([
+    spokePool.queryFilter(spokePool.filters.FilledRelay()),
+    spokePool.chainId(),
+  ]);
+  const lastEvent = events[events.length - 1];
+  if (lastEvent.args) {
+    return {
+      amount: lastEvent.args.amount,
+      totalFilledAmount: lastEvent.args.totalFilledAmount,
+      fillAmount: lastEvent.args.fillAmount,
+      repaymentChainId: Number(lastEvent.args.repaymentChainId),
+      originChainId: Number(lastEvent.args.originChainId),
+      relayerFeePct: lastEvent.args.relayerFeePct,
+      realizedLpFeePct: lastEvent.args.realizedLpFeePct,
+      depositId: lastEvent.args.depositId,
+      destinationToken: lastEvent.args.destinationToken,
+      relayer: lastEvent.args.relayer,
+      message: lastEvent.args.message,
+      depositor: lastEvent.args.depositor,
+      recipient: lastEvent.args.recipient,
+      updatableRelayData: lastEvent.args.updatableRelayData,
+      destinationChainId: Number(destinationChainId),
+    };
+  } else {
+    return null;
+  }
+}
+
+export function buildV2DepositForRelayerFeeTest(
+  amount: BigNumber,
+  tokenSymbol: string,
+  originChainId: string | number,
+  toChainId: string | number
+): V2Deposit {
+  const originToken = resolveContractFromSymbol(tokenSymbol, String(originChainId));
+  const destinationToken = resolveContractFromSymbol(tokenSymbol, String(toChainId));
+  expect(originToken).to.not.be.undefined;
+  expect(destinationToken).to.not.undefined;
+  if (!originToken || !destinationToken) {
+    throw new Error("Token not found");
+  }
+  return {
+    amount: toBN(amount),
+    depositId: bnUint32Max.toNumber(),
+    depositor: randomAddress(),
+    recipient: randomAddress(),
+    relayerFeePct: bnZero,
+    message: EMPTY_MESSAGE,
+    originChainId: 1,
+    destinationChainId: 10,
+    quoteTimestamp: getCurrentTime(),
+    originToken,
+    destinationToken,
+    realizedLpFeePct: bnZero,
+  };
+}
+
+/**
+ * Find the amount filled for a deposit at a particular block.
+ * @param spokePool SpokePool contract instance.
+ * @param relayData Deposit information that is used to complete a fill.
+ * @param blockTag Block tag (numeric or "latest") to query at.
+ * @returns The amount filled for the specified deposit at the requested block (or latest).
+ */
+export async function relayFilledAmount(
```

Comment:
I'm OK to remove `relayFilledAmount()` and `findFillBlock()` for now - I will ressurrect them in v3 versions.

Reply:
Or we can just nuke the v2-specific parts of `relayFilledAmount()` and make it v3-specific, since it _should_ already work.